v1.2/create-sequence.md

```diff
@@ -33,21 +35,92 @@ table td:first-child {
-`MAXVALUE` |
-`START` |
-`CYCLE` |
+`seqname` | The name of the sequence to create, which must be unique within its database and follow these [identifier rules](keywords-and-identifiers.html#identifiers). When the parent database is not set as the default, the name must be formatted as `database.name`.
+`INCREMENT` | The value by which the sequence is incremented. A negative number creates a descending sequence. A positive number creates an ascending sequence.<br><br>**Default:** `1`
+`MINVALUE` | The minimum value of the sequence. Default values apply if not specified or if you enter `NO MINVALUE`.<br><br>**Default for ascending:** `1` <br><br>**Default for descending:** `MININT`
+`MAXVALUE` | The maximum value of the sequence. Default values apply if not specified or if you enter `NO MAXVALUE`.<br><br>**Default for ascending:** `MAXINT` <br><br>**Default for descending:** `-1`
+`START` | The first value of the sequence. <br><br>**Default for ascending:** `1` <br><br>**Default for descending:** `-1`
+`CYCLE` | The sequence will wrap around when the sequence value hits the maximum or minimum value. If `NO CYCLE` is set, the sequence will not wrap.
 
 ## Sequence Functions
 
-Link to relevant functions.
+We support the following [SQL sequence functions](/functions-and-operators.html#sequence-functions):
+
+- `nextval('seqname')`
+- `currval('seqname')`
+- `lastval()`
+- `setval('seqname', value, is_called)`
 
 ## Examples
 
-### Example 1
+### Create a Sequence with Default Settings
+
+In this example, we create a sequence with default settings.
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE SEQUENCE customer_seq;
+~~~
+
+### Create a Sequence with User-Defined Settings
+
+In this example, we create a sequence that starts at 1000 and descends in increments of 2.
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE SEQUENCE desc_customer_list START 1000 INCREMENT -2;
+~~~
+
+### Create a Table with a Sequence
+
+In this example, we create a table using the sequence we created in the first example as the table's primary key.
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TABLE customer_list (
+  id INT PRIMARY KEY DEFAULT nextval('customer_seq'),
+  customer string,
+  address string
+  );
+~~~
+
+Insert a few records to see the sequence.
+
+{% include copy-clipboard.html %}
+~~~ sql
+> INSERT INTO customer_list (customer, address) VALUES ('Lauren', '123 Main Street'), ('Jesse', '456 Broad Ave'), ('Amruta', '9876 Green Parkway');
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SELECT * FROM customer_list;
+~~~
+~~~
++----+----------+--------------------+
+| id | customer |      address       |
++----+----------+--------------------+
+|  1 | Lauren   | 123 Main Street    |
+|  2 | Jesse    | 456 Broad Ave      |
+|  3 | Amruta   | 9876 Green Parkway |
++----+----------+--------------------+
+~~~
+
+### View the Current Value of a Sequence
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SELECT currval('customer_seq');
```

Comment:
Not-so-fun fact: this doesn't just straightforwardly return the current value, as one would expect. It returns the last value that _you got for this sequence in the current session_. This means that:

- If you fire up a `cockroach sql` and run `currval` before running any `nextval`s, it errors out: `currval of sequence "seq" is not yet defined in this session`
- if you call `currval` and see 5, then other users of the DB increment it to 10, and then you call `currval` again, you'll still see 5

Why doesn't it just get the current value? Because we have to mimic PG. Why does PG do this? I don't know. How do you actually get the current value? `SELECT * FROM my_sequence`, which isn't implemented yet.

Reply:
updated